node_app/server.js

```diff
@@ -52,101 +50,71 @@ app.post("/Organization", (req, res) => {
       res.json(response.data);
     })
     .catch((e) => res.send(e));
+});
 
 app.post("/Patient", (req, res) => {
-  let obj = req.body;
-  for (patient of obj.Patient) {
-    let resource = {
-      resourceType: "Patient",
-      name: [
-        {
-          family: patient.family,
-          given: [patient.given],
-          suffix: [patient.suffix],
-        },
-      ],
-      telecom: [
-        // add later
-      ],
-      gender: patient.gender,
-      birthDate: patient.birthDate,
-      address: [
-        {
-          line: [patient.address.line],
-          city: patient.address.city,
-          state: patient.address.state,
-          postalCode: patient.address.postalCode,
-          country: patient.address.country,
-        },
-      ],
-      photo: [
-        {
-          url: "", // add later
-          title: `Photo of ${patient.given} ${patient.family} ${patient.suffix}`,
-        },
-      ],
-      contact: [
-        {
-          relationship: [
-            {
-              coding: [
-                {
-                  // unsure
-                  system: "http://terminology.hl7.org/CodeSystem/v2-0131",
-                  code: patient.contact.relationship,
-                  display: patient.contact.relationship,
-                },
-              ],
-            },
-          ],
-          name: {
-            family: patient.contact.family,
-            given: [patient.contact.given],
-          },
-          telecom: [
-            {
-              system: "phone",
-              value: patient.contact.phone.value,
-              use: patient.contact.phone.use,
-            },
-          ],
-        },
-      ],
-      communication: [
-        {
-          language: {
-            text: patient.language,
-          },
-          preferred: true,
-        },
-      ],
+  createPatient(req, res).catch((e) =>
+    res.status(400).json({
+      error: e.response ? e.response.data : e.message,
+    })
+  );
+});
+
+async function createPatient(req, res) {
+  let patients = req.body.Patient;
+  let patient = patients.shift();
+  let promises = [];
+  let response = { Patient: [] };
+
+  // resource for head of household
+  if (patient) {
+    let related = { resourceType: "RelatedPerson" };
+    let result = await createPatient_(patient, related);
+    response.Patient.push(result.data);
+
+    // update related
+    let related_id = result.data.link[0].other.reference.split("/")[1];
+    resource = RelatedPerson.toFHIR({
+      patient: `Patient/${result.data.id}`,
+      relationship: patient.relationship,
+    });
+    resource.id = related_id;
+    await axios.put(`${base}/RelatedPerson/${related_id}`, resource, headers);
+  }
```

Comment:
You could probably have an else block that just quits early with an exception or something (depends on if we want to consider calling createPatient with no patients a valid workflow and how robust against errors we want the broker to be) if patients.shift() doesn't return anything.

Reply:
Currently, it isn't handled explicitly, but the server returns
```
{
    "Patient": []
}
```
as if it quits early. Do we want to throw an exception?

Reply:
It doesn't quit early, but your logic is well-guarded by conditionals to prevent weird stuff from happening. That said, it will probably be safer in the long run if you have a "return-early-if" check near the top to validate inputs, just in case this code is refactored and that is opened up.